src/unread/unreadModel.js

```diff
@@ -40,6 +40,13 @@ export const getUnreadHuddles = (state: GlobalState): UnreadHuddlesState => stat
 
 export const getUnreadMentions = (state: GlobalState): UnreadMentionsState => state.unread.mentions;
 
+/** The total number of unreads in the given topic. */
+export const getUnreadCountForTopic = (
+  unread: UnreadState,
```

Comment:
Is it fine for a selector to take a substate (like `UnreadState`) as the input state, or should all selectors start with `GlobalState`?

Reply:
Yeah, perhaps calling this a "selector" is confusing and I should label it something else. Maybe "getter"?

The reason I want it taking just the unread state is so that we can conveniently use it in a context like #4635, in code where we have a `BackgroundData` style of object but don't have the whole Redux state.

More generally, a direction I'm interested in taking our data-model code is toward having each subsystem's model be more of a self-contained thing with its own identity and interface, vs. the status quo which kind of tries (but inconsistently so, because this doesn't really work if you try to push it hard) to treat them all as undifferentiated implementation details of the global state. For example, we'd have the whole "unreads" model as one such subsystem. So having a getter like this directly on that model's state is a small experiment in that direction.

Reply:
I think the name "selector" is probably fine, and the use case makes sense. Thanks for explaining, and I'm excited to see our data-model code structured more sensibly. 🙂 

We'd call both `rootReducer` and `messagesReducer` "reducers", and I think it's fine to call this function (and those that take `GlobalState`) selectors.